nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -349,6 +380,82 @@ in {
         Windows can unseal the encryption key.
       '';
     };
+
+    windows = mkOption {
+      default = { };
+      description = ''
+        Make Windows bootable from systemd-boot. This option is not necessary when Windows and
+        NixOS use the same EFI System Partition (ESP). In that case, Windows will automatically be
+        detected by systemd-boot.
+
+        However, if Windows is installed on a separate drive or ESP, you can use this option to add
+        a menu entry for each installation manually.
+
+        The attribute name is used for the title of the menu entry and internal file names.
+      '';
+      example = literalExpression ''
+        {
+          "10".efiDeviceHandle = "HD0c3";
+          "11-ame" = {
+            title = "Windows 11 Ameliorated Edition";
+            efiDeviceHandle = "HD0b1";
+          };
+          "11-home" = {
+            title = "Windows 11 Home";
+            efiDeviceHandle = "FS1";
+            sortKey = "z_windows";
+          };
+        }
+      '';
+      type = types.attrsOf (types.submodule ({ config, ... }: {
+        options = {
+          efiDeviceHandle = mkOption {
+            type = types.str;
+            example = "HD1b3";
+            description = ''
+              The device handle of the EFI System Partition (ESP) where the Windows bootloader is
+              located. This is the device handle that the EDK2 UEFI Shell uses to load the
+              bootloader.
+
+              To find this handle, follow these steps:
+              1. Set {option}`boot.loader.systemd-boot.edk2-uefi-shell.enable` to `true`
+              2. Run `nixos-rebuild boot`
+              3. Reboot and select "EDK2 UEFI Shell" from the systemd-boot menu
+              4. Run `map -c` to list all consistent device handles
+              5. For each device handle (for example, `HD0c1`), run `ls HD0c1:\EFI`
+              6. If the output contains the directory `Microsoft`, you might have found the correct device handle
+              7. Run `HD0c1:\EFI\Microsoft\Boot\Bootmgfw.efi` to check if Windows boots correctly
+              8. If it does, this device handle is the one you need (in this example, `HD0c1`)
+
+              This option is required, there is no useful default.
+            '';
+          };
+
+          title = mkOption {
+            type = types.str;
+            example = "Michaelsoft Binbows";
+            default = "Windows ${config._module.args.name}";
+            defaultText = ''attribute name of this entry, prefixed with "Windows "'';
+            description = ''
+              The title of the boot menu entry.
+            '';
+          };
+
+          sortKey = mkOption {
+            type = types.str;
+            default = "o_windows_${config._module.args.name}";
```

Comment:
Nit: FYI:

```suggestion
      type = types.attrsOf (types.submodule ({ config, name, ... }: {
        options = {
          efiDeviceHandle = mkOption {
            type = types.str;
            example = "HD1b3";
            description = ''
              The device handle of the EFI System Partition (ESP) where the Windows bootloader is
              located. This is the device handle that the EDK2 UEFI Shell uses to load the
              bootloader.

              To find this handle, follow these steps:
              1. Set {option}`boot.loader.systemd-boot.edk2-uefi-shell.enable` to `true`
              2. Run `nixos-rebuild boot`
              3. Reboot and select "EDK2 UEFI Shell" from the systemd-boot menu
              4. Run `map -c` to list all consistent device handles
              5. For each device handle (for example, `HD0c1`), run `ls HD0c1:\EFI`
              6. If the output contains the directory `Microsoft`, you might have found the correct device handle
              7. Run `HD0c1:\EFI\Microsoft\Boot\Bootmgfw.efi` to check if Windows boots correctly
              8. If it does, this device handle is the one you need (in this example, `HD0c1`)

              This option is required, there is no useful default.
            '';
          };

          title = mkOption {
            type = types.str;
            example = "Michaelsoft Binbows";
            default = "Windows ${name}";
            defaultText = ''attribute name of this entry, prefixed with "Windows "'';
            description = ''
              The title of the boot menu entry.
            '';
          };

          sortKey = mkOption {
            type = types.str;
            default = "o_windows_${name}";
```

Reply:
Ahh, that's much nicer, I didn't know this was an option. Fixed it!